addons/website_event_sale/controllers/main.py

```diff
@@ -23,13 +23,24 @@ def _create_attendees_from_registration_post(self, event, registration_data):
         if not any(info.get('event_ticket_id') for info in registration_data):
             return super()._create_attendees_from_registration_post(event, registration_data)
 
+        ticket_ids = {data['event_ticket_id'] for data in registration_data if data.get('event_ticket_id')}
+        registration_tickets = request.env['event.event.ticket'].sudo().browse(ticket_ids)
+        free_ticket_ids = set(registration_tickets.filtered(lambda ticket: ticket.price == 0).ids)
+
+        free_registration_data = [data for data in registration_data if data.get('event_ticket_id') in free_ticket_ids]
+        priced_registration_data = [data for data in registration_data if data not in free_registration_data]
+
+        if len(free_registration_data) == len(registration_data):
```

Comment:
Can't you replace lines 27-33 by these 2 lines:
```
for ticket in request.env['event.event.ticket'].sudo().browse(ticket_ids):
  if ticket.price <> 0:
    ...
```

Reply:
Hello @fpodoo 

For the first point, 
I tried but it doesn't work due to the following reasons,
we need to filter out the registration data based on the ticket price (with and without price) because we are preparing the `ticket_data` in 43 to 47 to update the cart from 49 to 56
and to add the keys to link it with the sale order in 58 to 62 for our use case we don't require all three steps for tickets without the price. These steps required only registration data which contains the paid tickets.
Or am I missing something?

For the second point
I have tested that flow and it seems working fine.

Thanks!!